metrics/lib/common.bash

```diff
@@ -73,6 +73,29 @@ function check_cmds()
 	done
 }
 
+# This function performs a docker pull on the image names
+# passed in (notionally as 'about to be used'), to ensure
+#  - that we have the most upto date images
+#  - that any pull/refresh time (for a first pull) does not
+#    happen during the test itself.
+#
+# The image list can be received standalone or as an array, e.g.
+#
+# images=(“img1” “img2”)
+# check_imgs "${images[@]}"
+function check_images()
+{
+	local img req_images=( "$@" )
+	for img in "${req_images[@]}"; do
+		echo "docker pull'ing: $img"
+		if ! docker pull "$img" > /dev/null 2>&1; then
```

Comment:
How about we drop the `stdout` redirect so that the tests always log the image revision (which would potentially help with problem diagnosis).

Reply:
Agreed - I'd considered that more for the 'show me what went wrong' factor as well - will update.

Reply:
updated and pushed - there was no trivial way to just get a nice synopsis from `docker pull`, so I've removed all redirect and now it is a touch verbose. Let's see how that pans out - and if we have to in the future we can filter it with some greppy'ness maybe.